datafusion/core/src/physical_optimizer/tests/aggregate_statistics.rs

```diff
@@ -15,292 +15,19 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Utilizing exact statistics from sources to avoid scanning data
```

Comment:
I worry that putting these tests  in `datafusion/core/src/physical_optimizer/tests/aggregate_statistics.rs` will make them harder to find becase:
1. It is in a different crate than being tested
2. It is in a non standard testing location

As I understand it, these tests use code in other modules that physical optimizer doesn't depend on (like `functions-aggregate`


Thus, I would like to suggest we put these tests https://github.com/apache/datafusion/tree/main/datafusion/core/tests


Perhaps something like `physical_optimizer_integration.rs` to mirror https://github.com/apache/datafusion/blob/main/datafusion/core/tests/optimizer_integration.rs